OpenRA.Game/FieldLoader.cs

```diff
@@ -46,7 +46,7 @@ public MissingFieldsException(string[] missing, string header = null, string hea
 
 		public static Func<string, Type, string, object> InvalidValueAction = (s, t, f) =>
 		{
-			throw new InvalidOperationException("FieldLoader: Cannot parse `{0}` into `{1}.{2}` ".F(s, f, t));
+			throw new YamlException("FieldLoader: Cannot parse `{0}` into `{1}.{2}` ".F(s, f, t));
```

Comment:
!

If you want to change this, please check for any code that does a ```catch (InvalidOperationException)``` and adjust that too.